### PR TITLE
DOCS-5130, DOCS-5170 Log Explorer Search Edits

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Refreshes the Log Explorer and Search Logs doc pages (removes alias for `logs/explorer/search` that was redirecting to the [Search Syntax page](https://docs.datadoghq.com/logs/explorer/search_syntax/)), cleanup legacy links for `logs/explorer/group`, updates syntax requirements about escaping spaces with the `\` character.

### Motivation

DOCS-5130, DOCS-5170

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/logs/explorer/search_syntax.md [hunk 1]
```diff
@@ -52,9 +50,13 @@ Use the search bar's autocomplete feature to complete your query using:
 
 Clear error states inform you which part of the query contains syntax errors and how to remediate them. For example, if you input the query `service:` with no value, the message "Missing value in key:value pair" is displayed when you hover over the query.
 
-## Escaping of special characters
+## Escape special characters and spaces
 
-The following characters are considered special: `+` `-` `=` `&&` `||` `>` `<` `!` `(` `)` `{` `}` `[` `]` `^` `"` `“` `”` `~` `*` `?` `:` `\`, and `/` require escaping with the `\` character. You can’t search for special characters in a log message. However, you can search for special characters when they’re inside of an attribute. To search for special characters, parse them into an attribute with the [grok parser][1], and then search for logs that contain the attribute.
+The following characters are considered special: `+` `-` `=` `&&` `||` `>` `<` `!` `(` `)` `{` `}` `[` `]` `^` `"` `“` `”` `~` `*` `?` `:` `\`, while spaces and `/` require escaping with the `\` character.
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:55]:
```suggestion
The following characters are considered special: `+` `-` `=` `&&` `||` `>` `<` `!` `(` `)` `{` `}` `[` `]` `^` `"` `“` `”` `~` `*` `?` `:` `\`, and spaces require escaping with the `\` character. 
```

Comment (maycmlee) [reply content/en/logs/explorer/search_syntax.md:55]: I believe all special characters _and_ spaces need to be escaped with `\`.

File: content/en/logs/explorer/search_syntax.md [hunk 2]
```diff
@@ -52,9 +50,13 @@ Use the search bar's autocomplete feature to complete your query using:
 
 Clear error states inform you which part of the query contains syntax errors and how to remediate them. For example, if you input the query `service:` with no value, the message "Missing value in key:value pair" is displayed when you hover over the query.
 
-## Escaping of special characters
+## Escape special characters and spaces
 
-The following characters are considered special: `+` `-` `=` `&&` `||` `>` `<` `!` `(` `)` `{` `}` `[` `]` `^` `"` `“` `”` `~` `*` `?` `:` `\`, and `/` require escaping with the `\` character. You can’t search for special characters in a log message. However, you can search for special characters when they’re inside of an attribute. To search for special characters, parse them into an attribute with the [grok parser][1], and then search for logs that contain the attribute.
+The following characters are considered special: `+` `-` `=` `&&` `||` `>` `<` `!` `(` `)` `{` `}` `[` `]` `^` `"` `“` `”` `~` `*` `?` `:` `\`, while spaces and `/` require escaping with the `\` character. 
+
+Instead of searching for special characters in a log message, you can search for special characters when they are inside of an attribute.
```
Comment (maycmlee) [inline content/en/logs/explorer/search_syntax.md:57]: I would still mention that you can't search for special characters, just so it's clear.

**Review by maycmlee — COMMENTED** [review]
Left some feedback. Thanks for working on this!

File: content/en/logs/explorer/search_syntax.md (hunk 2 above)
Comment (alai97) [reply content/en/logs/explorer/search_syntax.md:57]:
```suggestion
You cannot search for special characters in a log message. You can search for special characters when they are inside of an attribute.
```

**Review by maycmlee — APPROVED** [review]
LGTM!

**Pull request merged**